src/site/content/en/blog/webassembly-threads/index.md

```diff
@@ -0,0 +1,472 @@
+all browsers with WebAssembly support. You can find which browsers support new WebAssembly features
+on the [webassembly.org roadmap](https://webassembly.org/roadmap/).
+
+To ensure that all users can load your application, you'll need to implement progressive enhancement
+by building two different versions of Wasm—one with multithreading support and one without it. Then
+load the supported version depending on feature detection results. To detect WebAssembly threads
+support at runtime, use [wasm-feature-detect
+library](https://github.com/GoogleChromeLabs/wasm-feature-detect) and load the module like this:
+
+```js
+import { threads } from 'wasm-feature-detect';
+
+const hasThreads = await threads();
+
+const module = await (
+  hasThreads
+    ? import('./module-with-threads.js')
+    : import('./module-without-threads.js')
+);
+
+// …now use `module` as you normally would
+```
+
+Now let's take a look at how to build a multithreaded version of the WebAssembly module.
+
+### C
+
+In C, particularly on Unix-like systems, the common way to use threads is via [POSIX
+Threads](https://en.wikipedia.org/wiki/POSIX_Threads) provided by the `pthread` library. Emscripten
+[provides an API-compatible implementation](https://emscripten.org/docs/porting/pthreads.html) of
+the `pthread` library built atop Web Workers, shared memory and atomics, so that the same code can
+work on the Web without changes.
+
+Let's take a look at an example:
+
+```c/2,17,19
+#include <stdio.h>
+#include <unistd.h>
+#include <pthread.h>
+
+void *thread_callback(void *arg)
+{
+    sleep(1);
+    printf("Inside the thread: %d\n", *(int *)arg);
+    return NULL;
+}
+
+int main()
+{
+    puts("Before the thread");
+
+    pthread_t thread_id;
+    int arg = 42;
+    pthread_create(&thread_id, NULL, thread_callback, &arg);
+
+    pthread_join(thread_id, NULL);
+
+    puts("After the thread");
+
+    return 0;
+}
+```
+
+Here the headers for the pthread library are included via `pthread.h`. You can also see a couple of
+crucial functions for dealing with threads.
+
+[`pthread_create`](https://man7.org/linux/man-pages/man3/pthread_create.3.html) will create a
+background thread. It takes a destination to store a thread handle in, some thread creation
+attributes (here not passing any, so it's just `NULL`), the callback to be executed in the new
+thread (here `thread_callback`), and an optional argument pointer to pass to that callback in case
+you want to share some data from the main thread - in this example we're sharing a pointer to a
+variable `arg`.
+
+[`pthread_join`](https://man7.org/linux/man-pages/man3/pthread_join.3.html) can be called later at
+any time to wait for the thread to finish the execution, and get the result returned from the
+callback. It accepts the earlier assigned thread handle as well as a pointer to store the result in
+- in this case I don't have any results, so I'm just passing `NULL`.
```

Comment:
This is not a rule we enforce strictly, but try to avoid speaking in first person in code instructions (both "we" and "I"). For example, you could rephrase like this: "In this case, there aren't any results so the function takes a `NULL` argument".

Reply:
Good point, thanks.